### DISCUSSION TITLE
Use of RtpsRelayOnly for Unicast no RTPS discovery solution

### DISCUSSION THREAD

**Discussion by 0neTX** [jmmorato/openddsharp#262] [Q&A] [answered]
I’m trying to set up three OpenDDSharp nodes using RtpsRelayOnly on the rtps.ini configuration file for a scenario where each node is both a publisher and a subscriber. All nodes have static, known IP addresses, and I need them to communicate using unicast and no RTPS discovery.

The configuration works perfectly on multicast mode, where the nodes can discover each other using broadcast discovery and exchange messages as expected. However, when switching to unicast, the nodes fail to establish the connection. I want to use a file with the IP addresses that are allowed on each peer for safety reasons.
I don’t think that the Relay is even starting to listen.

In unicast mode I’ve tried with firewalls down and using ICE, just in case, as stated on the official docs: [https://opendds.readthedocs.io/en/latest/devguide/internet_enabled_rtps.html#rtpsrelay](https://opendds.readthedocs.io/en/latest/devguide/internet_enabled_rtps.html#rtpsrelay)
And I’ve heard in a webinar that if ACE isn’t compiled with security enabled It can’t use RtpsRelay since its unsafe. I assumed that RtpsRelay was available by default.

If you could confirm any of that and if you need the .ini file or even the example project that I’m working on to give some advice I’ll be glad to provide it but it is similar to this discussion (with the RtpsRelay solution mentioned): [https://github.com/OpenDDS/OpenDDS/discussions/4654](https://github.com/OpenDDS/OpenDDS/discussions/4654)

Thanks.

**Comment by mitza-oci** [comment, marked as answer]
> And I’ve heard in a webinar that if ACE isn’t compiled with security enabled It can’t use RtpsRelay since its unsafe. I assumed that RtpsRelay was available by default.

... if *OpenDDS* isn't compiled with DDS Security enabled...

(doesn't relate to ACE)

> If you could confirm any of that and if you need the .ini file or even the example project that I’m working on to give some advice I’ll be glad to provide it but it is similar to this discussion (with the RtpsRelay solution mentioned): [OpenDDS/OpenDDS#4654](https://github.com/OpenDDS/OpenDDS/discussions/4654)

See https://github.com/OpenDDS/cloud-multicast-repeater referenced in that discussion.
